src/main/java/org/spongepowered/common/service/pagination/PaginationCalculator.java

```diff
@@ -218,82 +213,57 @@ int getWidth(Component text) {
-            styledPadding = this.withStyle(padding, text);
-            paddingLength = this.getWidth(styledPadding);
+            paddingLength = this.getWidth(padding);
         }
 
         //if we only need padding
         if (inputLength == 0) {
-            this.addPadding(padding, output, GenericMath.floor((double) LINE_WIDTH / paddingLength));
+            this.addPadding(padding, output, GenericMath.floor((double) PaginationCalculator.LINE_WIDTH / paddingLength));
         } else {
-            if(addSpaces) {
+            if (addSpaces) {
                 text = textWithSpaces;
                 inputLength = this.getWidth(textWithSpaces);
             }
 
-            int paddingNecessary = LINE_WIDTH - inputLength;
+            final int paddingNecessary = PaginationCalculator.LINE_WIDTH - inputLength;
 
-            int paddingCount = GenericMath.floor(paddingNecessary / paddingLength);
+            final int paddingCount = GenericMath.floor(paddingNecessary / paddingLength);
             //pick a halfway point
-            int beforePadding = GenericMath.floor(paddingCount / 2.0);
+            final int beforePadding = GenericMath.floor(paddingCount / 2.0);
             //Do not use ceil, this prevents floating point errors.
-            int afterPadding = paddingCount - beforePadding;
+            final int afterPadding = paddingCount - beforePadding - 1;
 
-            this.addPadding(styledPadding, output, beforePadding);
+            this.addPadding(padding, output, beforePadding);
             output.append(text);
-            this.addPadding(styledPadding, output, afterPadding);
+            this.addPadding(padding, output, afterPadding);
         }
 
         return this.finalizeBuilder(text, output);
     }
 
-    /**
-     * Gives the first text argument the style of the second.
-     *
-     * @param text The text to stylize
-     * @param styled The styled text
-     * @return The original text now stylized
-     */
-    private Component withStyle(Component text, Component styled) {
-        return text.style(styled.style());
-    }
-
-    /**
-     * Gives the first text argument the color of the second.
-     *
-     * @param text The text to color
-     * @param colored The colored text
-     * @return The original text now colored
-     */
-    private Component withColor(Component text, Component colored) {
-        return text.color(colored.color());
-    }
-
     /**
      * Finalizes the builder used in centering text.
      *
      * @param text The text to get the style from
      * @param build The work in progress text builder
      * @return The finalized, properly styled text.
      */
-    private Component finalizeBuilder(Component text,TextComponent.Builder build) {
-        return build.style(text.style()).build();
+    private Component finalizeBuilder(final Component text, final TextComponent.Builder build) {
+        return build.style(text.style().toBuilder().build()).build();
```

Comment:
Is there a reason for this `toBuilder().build()`? The following compiles fine:

```
final Component component = TextComponent.empty();
TextComponent.builder().style(component.style()).build();
```

I haven't run it though - if there is a problem that you've worked around, we need to report it to Adventure.

Reply:
Oh I didn't put a comment there to explain.
It seems to be needed to avoid StackOverflowExceptions that happen later for some reason. I mentioned it to kashike and AFAIK we're not sure why this happens.

Reply:
Is there an issue reported for this? I imagine it's probably on the Adventure end but it needs to be reported somewhere.

Once done, add a comment linking to that issue so if some poor soul wants more information (and perhaps wants to have a stab at fixing it), they can do so.

Reply:
Well I don't even know if the issue lies in the library or the SpongeCommon glue code. I should talk to kashike about it again I think.

Reply:
Oh, I just looked and I know exactly what's going on. It's basically the fact that the Minecraft `Style` object is being cached for a given Adventure `Style` object, but when you use Minecraft's `Style#setStyle` it also sets the parent of the child's style. In this case, the parent and the child have been supplied with the same Style reference, so when the parent style is set... it's itself.

Sponge error. We basically should not be caching mutable objects.

Reply:
https://github.com/SpongePowered/SpongeCommon/commit/44be61c0dbc79069f82c174b63d7a68aa254c1b7

Reply:
Amazing, thank you ! I hope this is the last force push I have to do for this PR.